### PR TITLE
[core] add HttpRequest and HttpResponse reprs

### LINKED ISSUE #16952: [core] add good reprs for our HttpRequest and HttpResponses

### PR DESCRIPTION
**Pull request by iscai-msft**
fixes #16952

### CODE REVIEW

File: sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py
Comment (iscai-msft) [inline sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py:270]: callout: I think this should be `self.headers` instead of `aiohttp_response.headers`, bc we made `self.headers` case insensitive

Comment (johanste) [reply sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py:270]: Technically, from a functional perspective, this shouldn't matter since `aiohttp_response.headers` is also case-insensitive.

Comment (iscai-msft) [reply sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py:270]: yeah, @lmazuel brought that up too. tbh, i'll just go ahead and delete it, it doesn't need to be there

File: sdk/core/azure-core/azure/core/pipeline/transport/_base.py
Comment (johanste) [inline sdk/core/azure-core/azure/core/pipeline/transport/_base.py:599]: Nitpick: I would exclude the content type key/value altogether in the repr if we don't know what it is (i.e. not add `, Content-Type: ` to the output...

**Review by kristapratico — APPROVED** [review]
Verbal confirmation of approval for this PR was given, I'm just approving so we can merge.

**Comment by iscai-msft** [comment]
thanks @kristapratico , @johanste and @lmazuel have already given their verbal approval

**Review by xiangyan99 — APPROVED** [review]

**Pull request merged**
